packages/react-hooks/src/useBoolean.ts

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import { useConstCallback } from './useConstCallback';
+
+/** Result returned by `useBoolean`. */
+export interface IUseBooleanResult {
```

Comment:
I'm still a huge fan of `const [isTrue, {setTrue, setFalse, toggle}] = useBoolean(false);`. Separating the returned value from the actions you can take on the value. Hooks typically return an array with 2 items, first item is the value, 2nd item is how to modify the value. Since there is more than 1 way to modify the value, wrap them up in a single object.

```jsx
const [value, changeValue] = useBoolean(false);

// changeValue.setTrue();
// changeValue.setFalse();
// changeValue.toggle();



```

Reply:
Another option is this `useToggle`

https://streamich.github.io/react-use/?path=/story/state-useboolean--docs

```jsx
<button onClick={toggle}>Toggle</button>
<button onClick={() => toggle(true)}>set ON</button>
<button onClick={() => toggle(false)}>set OFF</button>
```

Reply:
The useCounter suggestion is interesting. I don't like the other useToggle because it's helpful (for reducing boilerplate) in a lot of cases to have a premade callback specifically for setting to true or false, for passing as an onClick handler for example.